services/registration/src/routes/branch.ts

```diff
@@ -1,12 +1,19 @@
 import { asyncHandler } from "@api/common";
 import express from "express";
+import { FilterQuery } from "mongoose";
 
-import { BranchModel } from "../models/branch";
+import { Branch, BranchModel } from "../models/branch";
 
 export const branchRouter = express.Router();
 
 branchRouter.route("/").get(
   asyncHandler(async (req, res) => {
+    const filter: FilterQuery<Branch> = {};
+
+    if (req.query.hexathon) {
+      filter.hexathon = req.query.hexathon;
+    }
```

Comment:
You created the filter, but you didn't add it into the mongoose request. Can you please add it in and test it to ensure it works properly?

Reply:
Also, can you update the docs swagger to have a query property to filter by hexathon? You can look at the users request to see how the query parameter is added.